lib/nerves_hub_link/update_manager.ex

```diff
@@ -186,43 +191,42 @@ defmodule NervesHubLink.UpdateManager do
     # interrupt FWUP and let the task finish. After update and reboot, the
     # device will check-in and get an update message if it was actually new and
     # required
-    state
+    {:ignored, state}
   end
 
   defp maybe_update_firmware(%UpdateInfo{} = update_info, fwup_public_keys, %State{} = state) do
-    # Cancel an existing timer if it exists.
-    # This prevents rescheduled updates`
-    # from compounding.
-    state = maybe_cancel_timer(state)
-
     # possibly offload update decision to an external module.
     # This will allow application developers
     # to control exactly when an update is applied.
     # note: update_available is a behaviour function
     case state.fwup_config.update_available.(update_info) do
       :apply ->
-        start_fwup_stream(update_info, fwup_public_keys, state)
+        NervesHubLink.send_firmware_update_status(:updating)
+        {{:updating, 0}, start_fwup_stream(update_info, fwup_public_keys, state)}
 
       :ignore ->
-        state
+        NervesHubLink.send_firmware_update_status(:ignored)
+        Logger.info("[NervesHubLink] ignoring firmware update request")
+        {:ignored, %{state | status: :idle, previous_status: :ignored}}
 
-      {:reschedule, ms} ->
-        timer =
-          Process.send_after(self(), {:update_reschedule, update_info, fwup_public_keys}, ms)
-
-        Logger.info("[NervesHubLink] rescheduling firmware update in #{ms} milliseconds")
-        %{state | status: :update_rescheduled, update_reschedule_timer: timer}
-    end
-  end
+      {:reschedule, minutes, :minutes} ->
+        until =
+          DateTime.utc_now()
+          |> DateTime.add(minutes, :minute)
 
-  defp maybe_update_firmware(_, _, state), do: state
+        NervesHubLink.send_firmware_update_status(:reschedule, %{until: until})
+        Logger.info("[NervesHubLink] rescheduling firmware update in #{minutes} minutes")
+        {:rescheduled, %{state | status: :idle, previous_status: :rescheduled}}
```

Comment:
I don't understand the addition of `previous_status`. It seems to me that where it is used, `:status` is being set to `:idle`. So now we have a split of states that doesn't seem necessary? It seems saying something like `status: :rescheduled` is a valid state to sit in.

Reply:
My intention was to be able to see what the current status of the `UpdateManager` was, along with the result from the previous request.
eg. is it doing something or is it available to process an update, and what was the outcome of the last request.

I could remove `previous_status` and just use the new simplified statuses of `idle`, `{:updating, integer()}`, and `:applying`, and rely on message sending for the result of an update request 
(eg. `:ignored`, `:rescheduled`, `:successful`, and `{:error, String.t()}`)